src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

```diff
@@ -0,0 +1,73 @@
+package org.jenkinsci.plugins.ewm.casc;
+
+import hudson.ExtensionList;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.ewm.definitions.Disk;
+import org.jenkinsci.plugins.ewm.definitions.DiskPool;
+import org.jenkinsci.plugins.ewm.steps.ExwsAllocateStep;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertEquals;
+
+import org.jvnet.hudson.test.JenkinsRule;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Test for Configuration As Code Compatibility.
+ *
+ * @author Martin d'Anjou
+ */
+public class ConfigAsCodeTest {
+
+    @ClassRule public static JenkinsRule r = new JenkinsRule();
+
+    @Test
+    public void shouldSupportConfigurationAsCode() throws Exception {
+        URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
+        String config = resource.toString();
+        ConfigurationAsCode.get().configure(config);
+        // get the jenkins rule object.
+
+        ExwsAllocateStep.DescriptorImpl descriptor =  ExtensionList.lookupSingleton(ExwsAllocateStep.DescriptorImpl.class);
+        List<DiskPool> diskPools = descriptor.getDiskPools();
+        DiskPool diskPool = diskPools.get(0);
+
+        // assertion
+        assertThat(diskPool.getDiskPoolId(), is("diskpool1"));
+        assertThat(diskPool.getDisplayName(), is("diskpool1 display name"));
+        assertThat(diskPool.getDisks().get(0).getDiskId(), is("disk1"));
+        assertThat(diskPool.getDisks().get(0).getDisplayName(), is("disk one display name"));
+        assertThat(diskPool.getDisks().get(0).getMasterMountPoint(), is("/tmp"));
+    }
+
+    @Test
+    public void exportConfiguration() throws Exception {
+        Yaml yaml = new Yaml();
+
+        // get the CasC configure
+        ByteArrayOutputStream outstream = new ByteArrayOutputStream();
+        ConfigurationAsCode.get().export(outstream);
```

Comment:
I'm slightly confused how this is working, you aren't calling `.configure` in this test, if you print the maps out do they contain what you expect?

Reply:
I read the instructions on https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/PLUGINS.md   (scroll down to the bottom) and the example seems did not use any configure... And the exported map do match the yaml file ```unclassified`` entry. Not sure if i still need to call the .configure()

Reply:
on the first line it says: `/** Setup jenkins to use plugin */`

i.e. configure it appropriately first either through JCASC or java api

Weird, I wonder if it's because its a class rule and its reusing the same jenkins config as the previous config.

If you run just this test does it pass?

Reply:
Yes it passed (actually I checked the content of both files and the exported yaml file do contain the ```unclassified``` entry``` But i 'll double check it in case I misunderstood the console output.